### ISSUE TITLE
[🐛 Bug]: "400 Bad Request" response to any WebDriver.Execute call causes crash

### ISSUE THREAD

**Issue by 0x70b1a5** [SeleniumHQ/selenium#10865]
## How did you trigger this bug?

Run a test including any code that calls into `WebDriver.Execute`, such that the server responds with `400 Bad Request`. 

## What did you expect to happen? 

I expect the driver to retry failed requests until they succeed, rather than throwing an error about unparseable JSON. 

## What code is causing this issue? 

`/dotnet/src/webdriver/Response.cs#L147` 

### How can we reproduce the issue?

```shell
Unable to reproduce absent large server load, which is when it typically happens, AFAICT. I lack the skill to set up a server that would respond `400 Bad Request` to *any* `WebDriver.Execute`, but that is how I would test it.
```

### Relevant log output

```shell
Additional text encountered after finished reading JSON content: B. Path '', line 1, position 4.
 at Newtonsoft.Json.JsonTextReader.Read()
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
   at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
   at Newtonsoft.Json.JsonSerializer.Deserialize(JsonReader reader, Type objectType)
   at Newtonsoft.Json.JsonConvert.DeserializeObject(String value, Type type, JsonSerializerSettings settings)
   at Newtonsoft.Json.JsonConvert.DeserializeObject(String value, Type type, JsonConverter[] converters)
   at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value, JsonConverter[] converters)
   at OpenQA.Selenium.Response.FromJson(String value)
   at OpenQA.Selenium.Remote.HttpCommandExecutor.CreateResponse(HttpResponseInfo responseInfo)
   at OpenQA.Selenium.Remote.HttpCommandExecutor.Execute(Command commandToExecute)
   at OpenQA.Selenium.WebDriver.Execute(String driverCommandToExecute, Dictionary`2 parameters)
   at OpenQA.Selenium.WebDriver.InternalExecute(String driverCommandToExecute, Dictionary`2 parameters)
   at OpenQA.Selenium.WebElement.Execute(String commandToExecute, Dictionary`2 parameters)
   at OpenQA.Selenium.WebElement.get_Displayed()
```

### Operating System

Ubuntu

### Selenium version

C# Selenium.WebDriver 4.2.0

### What are the browser(s) and version(s) where you see this issue?

Chrome 102.0.5005.61 

### What are the browser driver(s) and version(s) where you see this issue?

ChromeDriver 

### Are you using Selenium Grid?

4.0.0

**Comment by diemol** [comment]
Retries should be handled by the user, with a wait strategy for example. However, we can give more feedback when we see a complete code snippet.

**Comment by titusfortner** [comment]
checking if an element is displayed executes a JavaScript atom in the browser.
Selenium expects the results to always be a JSON object with a `"value"` key.
For some reason the code is getting an invalid response (looks like it might just be an empty string)

@diemol does the grid do anything to make sure that the response is valid, or does it just pass back whatever the driver sends it?

As Diego said, though, you should be checking for whether an element is displayed using an explicit wait of some kind. If you're doing this, you can add the exception you are seeing to the list of Exception types the wait will retry:

```
wait.IgnoreExceptionTypes(typeof(NoSuchElementException), typeof(WhateverJSONParsingExceptionYouAreGetting));
```

**Comment by diemol** [comment]
The Grid passes the information along. However, I could double check things if there was a way to reproduce the issue.

**Comment by 0x70b1a5** [comment]
@titusfortner The code is not getting an empty string response, it is getting the string `400 Bad Request`. It parses this string as JSON character by character. `400 ` is valid JSON. While parsing, the next character `B` causes an exception because it is not a valid JSON character after an int. This exception is thrown all the way up the stack and halts test execution. 

I've not been able to reproduce this reliably, and I suspect that is partially due to the fact that it only happens in extremis. (Also partially due to the fact I need to keep hammering away at maintaining tests! 😂 ) We run tens of thousands of concurrent tests every day and all that I can tell you is that this started happening with Grid 4.

**Comment by 0x70b1a5** [comment]
@diemol If there were some way to spin up a custom server that responds to all requests with `400 Bad Request`, or, a custom web driver that makes requests with invalid data, I am certain that would reproduce the issue. It depends on which parts of the server respond `400`.

**Comment by titusfortner** [comment]
Oh, maybe this is related to #10805 
It doesn't solve the problem of why it is happening, but it will address why .NET thinks it is a successful response, when it is not.

I don't think the current PR actually fixes the problem, but I haven't tried it, yet.

**Comment by titusfortner** [comment]
The Selenium 3.x http library implementation would catch 400/500 bugs separately, but the new implementation passes them along for the user (in this case the Selenium library) to handle. Selenium 4.3 looks for correctly formatted w3c responses, and if it isn't in that format (e.g. a generic 400 error), it was marking the status as successful and then the code would fail later when it was trying to be used.

The code that will be released in 4.4 is going to properly add the status to the response. You are still going to get a `WebDriverException` for an unhandled error (because this is not a valid return value and Selenium can't do anything with it), but now it will give a better error message, including what command was attempting to be executed.

If you want to ignore the error as part of determining if an element is displayed, you will have to use an explicit wait and add `WebDriverException` with `IgnoreExceptionTypes()` as described above.

As to why you are getting a 400 error for this command in the first place, that is a chromedriver error of some kind, and there isn't anything more that we can do about that in the Selenium code.

Well, except for maybe tidying up the isDisplayed atom. We've talked about a larger [re-implementation of the atoms in TypeScript](https://github.com/SeleniumHQ/selenium/issues/9668#issuecomment-1072652972), but that may or may not help this.
